capstone/src/main/java/com/google/sps/servlets/clubs/EditClubServlet.java

```diff
@@ -0,0 +1,55 @@
+package com.google.sps.servlets;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/* Servlet that stores and returns data relating to clubs. */
+@WebServlet("/club-edit")
+public class EditClubServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+    String founderEmail = userService.getCurrentUser().getEmail();
+    ImmutableList<String> officers =
+        ImmutableList.copyOf(request.getParameter(Constants.OFFICER_PROP).split(","));
+
+    Query query =
+        new Query("Club")
+            .setFilter(
+                new FilterPredicate(
+                    Constants.PROPERTY_NAME,
+                    FilterOperator.EQUAL,
+                    request.getParameter(Constants.PROPERTY_NAME)));
+    Entity clubEntity = datastore.prepare(query).asSingleEntity();
```

Comment:
Could this ever fail? Do we need to do a check before we continue?

Reply:
I wouldn't think this could ever fail, since the user can only be modifying a club that already exists and we already check for unique club names, but I added and extra check just in case. Thanks!